util/design/secded_gen.py

```diff
@@ -317,6 +322,53 @@ def verify(cfgs):
     return error
 
 
+def ecc_encode(codetype: str, k: int, dataword: int) -> Tuple[int, int]:
+    log.info(f"Encoding ECC for {hex(dataword)}")
+
+    # first check to see if bit width is supported among configuration
+    with open(SECDED_CFG_PATH, 'r') as infile:
+        config = hjson.load(infile)
```

Comment:
nit: can replace with `config = hjson.load(SECDED_CFG_PATH.open())`

Reply:
ah looks good, should this still be bracketed with a try/except?

Reply:
Yes, thats more _pythonic_ from what I understand. [Specifically, the style of _Easier to Ask for Forgiveness than Permission_ (EAFP) is often considered more _pythonic_ than the _Look Before You Leap_ (LBYL) style.](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/
)